lib/class-wp-rest-menus-controller.php

```diff
@@ -431,6 +480,60 @@ public function delete_item( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Returns the value of a menu's auto_add
+	 *
+	 * @param int $menu_id The menu id to update the location form.
+	 *
+	 * @return bool The value of auto_add.
+	 */
+	function get_menu_auto_add( $menu_id ) {
+		$nav_menu_option = (array) get_option( 'nav_menu_options', array( 'auto_add' => array() ) );
+		$check = in_array( $menu_id, $nav_menu_option['auto_add'], true );
+		
+		return $check;
+	}
+
+	/**
+	 * Updates the menu's auto add from a REST request.
+	 *
+	 * @param int             $menu_id The menu id to update the location form.
+	 * @param WP_REST_Request $request The request object with menu and locations data.
+	 *
+	 * @return bool True if the auto update was successfully updated.
+	 */
+	function handle_auto_add( $menu_id, $request ) {
+		if ( ! isset( $request['auto_add'] ) ) {
+			return true;
+		}
+
+		$nav_menu_option = (array) get_option( 'nav_menu_options', array( 'auto_add' => array() ) );
+
+		if ( ! isset( $nav_menu_option['auto_add'] ) ) {
+			$nav_menu_option['auto_add'] = array();
+		}
+
+		$auto_add = $request['auto_add'];
+
+		if ( $auto_add ) {
+			if ( ! in_array( $menu_id, $nav_menu_option['auto_add'], true ) ) {
+				$nav_menu_option['auto_add'][] = $menu_id;
+			}
+		} else {
+			$key = array_search( $menu_id, $nav_menu_option['auto_add'], true );
+			if ( false !== $key ) {
+				unset( $nav_menu_option['auto_add'][ $key ] );
+			}
+		}
```

Comment:
```suggestion
		$i = array_search( $menu_id, $nav_menu_options['auto_add'], true );

		if ( $auto_add && false === $i ) {
			array_push( $nav_menu_options['auto_add'], $menu_id );
		} elseif ( ! $auto_add && false !== $i ) {
			array_splice( $nav_menu_options['auto_add'], $i, 1 );
		}
```

Reply:
What about this change @TimothyBJacobs

Reply:
Looks good. I'd just `$nav_menu_options['auto_add'][]` instead of using `array_push`.